### PR TITLE
TRT-1613: Bubble up error for getting alert disruption data

### PR DESCRIPTION
**Pull request by DennisPeriquet**
When the [scripting used in the automation](https://github.com/openshift/release/blob/ef1047161807a90062a6a596ae58b0de712c1027/ci-operator/jobs/infra-periodics.yaml#L2557-L2601) runs, we might get an error when fetching disruption data for alerts. Here's an example of a recently [failed run](https://prow.ci.openshift.org/view/gs/test-platform-results/logs/periodic-update-origin-disruption-alert-data/1788735954228350976) (which succeeded on a retry 2 days later):

```
Updating alerts data
Using target release: 4.17, previous release: 4.16
time="2024-05-10T01:03:24.825Z" level=fatal msg="Command failed" error="Failed while attempting to read Alert Historical Data"
```

If we every fail again, it would be nice to know what failed. This PR ensures that captured errors are "bubbled" up to the caller where they can be put into the final error message; this will help us know what operation had the error.

### CODE REVIEW

**Comment by DennisPeriquet** [comment]
will retry the CI later when [build09 autoscaler problems](https://redhat-internal.slack.com/archives/CBN38N3MW/p1715675319033009?thread_ts=1715674736.072099&cid=CBN38N3MW) are addressed.

File: pkg/jobrunaggregator/jobrunhistoricaldataanalyzer/analyzer.go [hunk 1]
```diff
@@ -54,9 +54,9 @@ func (o *JobRunHistoricalDataAnalyzerOptions) Run(ctx context.Context) error {
 
 	switch {
 	case o.newFile == "" && o.dataType == "alerts":
-		newHistoricalData = o.getAlertData(ctx)
+		newHistoricalData, err = o.getAlertData(ctx)
 		if newHistoricalData == nil {
-			return fmt.Errorf("Failed while attempting to read Alert Historical Data")
+			return fmt.Errorf("failed while attempting to read Alert Historical Data: %w", err)
```
Comment (stbenjam) [inline pkg/jobrunaggregator/jobrunhistoricaldataanalyzer/analyzer.go:59]: `errors.WithMessage(err, "failed while attempting to read Alert Historical Data")`  is the new way to do things

Comment (DennisPeriquet) [reply pkg/jobrunaggregator/jobrunhistoricaldataanalyzer/analyzer.go:59]: noted.  and thanks!

**Pull request merged**